media-gfx/yacreader/yacreader-9.5.0.recipe

```diff
@@ -66,15 +67,26 @@ defineDebugInfoPackage yacreader$secondaryArchSuffix \
 
 BUILD()
 {
-	qmake PREFIX=$appsDir/YACReader MANDIR=$manDir DOCDIR=$docDir .
+	qmake \
+		PREFIX="$appsDir/YACReader" \
+		MANDIR="$manDir" \
+		DOCDIR="$docDir" \
+		BINDIR="$appsDir"/YACReader \
+		LIBDIR="$appsDir"/YACReader/lib \
+		DATADIR="$appsDir"/YACReader/data \
+		.
 	make $jobArgs
 }
 
 INSTALL()
 {
-	mkdir -p $appsDir/YACReader
+	mkdir -p "$appsDir"/YACReader
 	make install
 
+	# Cleanup
+	mv "$appsDir"/YACReader/bin/* "$appsDir"/YACReader
```

Comment:
I kicked a build on x86_gcc2 with `hp -S yacreader_x86` and it seems the compilation is OK but there's something wrong at the install phase: `mv: cannot stat '/packages/yacreader_x86-9.5.0-1/.self/apps/YACReader/bin/*': No such file or directory`
<Details>

```
cd YACReaderLibrary/ && ( test -e Makefile || /boot/system/bin/x86/qmake -o Makefile /sources/yacreader-9.5.0/YACReaderLibrary/YACReaderLibrary.pro PREFIX=/packages/yacreader_x86-9.5.0-1/.self/apps/YACReader MANDIR=/packages/yacreader_x86-9.5.0-1/.self/documentation/man DOCDIR=/packages/yacreader_x86-9.5.0-1/.self/documentation/packages/yacreader_x86 BINDIR=/packages/yacreader_x86-9.5.0-1/.self/apps/YACReader LIBDIR=/packages/yacreader_x86-9.5.0-1/.self/apps/YACReader/lib DATADIR=/packages/yacreader_x86-9.5.0-1/.self/apps/YACReader/data ) && make -f Makefile install
make[1]: Entering directory '/sources/yacreader-9.5.0/YACReaderLibrary'
/boot/system/bin/x86/qmake -install qinstall -exe /sources/yacreader-9.5.0/YACReaderLibrary/YACReaderLibrary /packages/yacreader_x86-9.5.0-1/.self/apps/YACReader/YACReaderLibrary
/boot/system/bin/x86/qmake -install qinstall /sources/yacreader-9.5.0/release/server /packages/yacreader_x86-9.5.0-1/.self/apps/YACReader/data/data/server
/boot/system/bin/x86/qmake -install qinstall /sources/yacreader-9.5.0/release/languages/yacreaderlibrary_de.qm /packages/yacreader_x86-9.5.0-1/.self/apps/YACReader/data/data/languages/yacreaderlibrary_de.qm
/boot/system/bin/x86/qmake -install qinstall /sources/yacreader-9.5.0/release/languages/yacreaderlibrary_es.qm /packages/yacreader_x86-9.5.0-1/.self/apps/YACReader/data/data/languages/yacreaderlibrary_es.qm
/boot/system/bin/x86/qmake -install qinstall /sources/yacreader-9.5.0/release/languages/yacreaderlibrary_fr.qm /packages/yacreader_x86-9.5.0-1/.self/apps/YACReader/data/data/languages/yacreaderlibrary_fr.qm
/boot/system/bin/x86/qmake -install qinstall /sources/yacreader-9.5.0/release/languages/yacreaderlibrary_nl.qm /packages/yacreader_x86-9.5.0-1/.self/apps/YACReader/data/data/languages/yacreaderlibrary_nl.qm
/boot/system/bin/x86/qmake -install qinstall /sources/yacreader-9.5.0/release/languages/yacreaderlibrary_pt.qm /packages/yacreader_x86-9.5.0-1/.self/apps/YACReader/data/data/languages/yacreaderlibrary_pt.qm
/boot/system/bin/x86/qmake -install qinstall /sources/yacreader-9.5.0/release/languages/yacreaderlibrary_ru.qm /packages/yacreader_x86-9.5.0-1/.self/apps/YACReader/data/data/languages/yacreaderlibrary_ru.qm
/boot/system/bin/x86/qmake -install qinstall /sources/yacreader-9.5.0/release/languages/yacreaderlibrary_tr.qm /packages/yacreader_x86-9.5.0-1/.self/apps/YACReader/data/data/languages/yacreaderlibrary_tr.qm
/boot/system/bin/x86/qmake -install qinstall /sources/yacreader-9.5.0/YACReaderLibrary.1 /packages/yacreader_x86-9.5.0-1/.self/documentation/man/man1/YACReaderLibrary.1
make[1]: Leaving directory '/sources/yacreader-9.5.0/YACReaderLibrary'
mv: cannot stat '/packages/yacreader_x86-9.5.0-1/.self/apps/YACReader/bin/*': No such file or directory
```

</Details>

BTW, it would be nice to choose a better layout for the destination directories. Right now we have:
* these runtimes and libraries:
```diff
-apps/YACReader/YACReader
-apps/YACReader/YACReaderLibrary
```
* and these directories for the data files (other than the man file and the changelog):
```diff
-apps/YACReader/data/data/
-apps/YACReader/data/data/languages/
-apps/YACReader/data/data/server/
-apps/YACReader/data/data/server/docroot/
-apps/YACReader/data/data/server/docroot/css/
-apps/YACReader/data/data/server/docroot/images/
-apps/YACReader/data/data/server/templates/
```

It would be better, imho, to have:
* the runtimes as:
```diff
+apps/YACReader
+apps/YACReaderLibrary
```
* and these directories for the data files (other than the man file and the changelog):
```diff
+data/YACReader/
+data/YACReader/languages/
+data/YACReader/server/
+data/YACReader/server/docroot/
+data/YACReader/server/docroot/css/
+data/YACReader/server/docroot/images/
+data/YACReader/server/templates/
```
Well, an even better layout for the runtimes and libraries would be:
```diff
+apps/YACReader
+libs/x86/YACReader/YACReaderLibrary
```
on x86 secondary arch, and
```diff
+apps/YACReader
+libs/YACReader/YACReaderLibrary
```
on x86_64.
I guess this can be achieved with the appropriate arguments passed to qmake and then moving the YACReader runtime to `$appsDir`.
I think this would work:
```
BUILD()
{
	qmake \
		PREFIX="$prefix" \
		MANDIR="$manDir" \
		DOCDIR="$docDir" \
		BINDIR="$libDir"/YACReader \
		DATADIR="$dataDir"/YACReader \
		.
	make $jobArgs
}

INSTALL()
{
	make install
	mkdir -p "$appsDir"
	mv "$libDir"/YACReader/YACReader "$appsDir"
```

Reply:
Oops, I missed your last message before sending the one above. Let me suggest a better diff for `INSTALL`.

Reply:
I've just kicked a build with these changes:
```diff
 BUILD()
 {
 	qmake \
-		PREFIX="$appsDir/YACReader" \
+		PREFIX="$prefix" \
 		MANDIR="$manDir" \
 		DOCDIR="$docDir" \
-		BINDIR="$appsDir"/YACReader \
-		LIBDIR="$appsDir"/YACReader/lib \
-		DATADIR="$appsDir"/YACReader/data \
+		BINDIR="$appsDir" \
+		DATADIR="$dataDir"/YACReader \
 		.
 	make $jobArgs
 }
 
 INSTALL()
 {
-	mkdir -p "$appsDir"/YACReader
 	make install
-
-	# Cleanup
-	mv "$appsDir"/YACReader/bin/* "$appsDir"/YACReader
-	rm -rf "$appsDir"/YACReader/bin
 
```
and
```diff
 	addResourcesToBinaries yacreader.rdef \
-		"$appsDir"/YACReader/YACReader
+		"$appsDir"/YACReader
 	addResourcesToBinaries yacreader_library.rdef \
-		"$appsDir"/YACReader/YACReaderLibrary
+		"$appsDir"/YACReaderLibrary
 
-	addAppDeskbarSymlink "$appsDir"/YACReader/YACReader
-	addAppDeskbarSymlink "$appsDir"/YACReader/YACReaderLibrary
+	addAppDeskbarSymlink "$appsDir"/YACReader
+	addAppDeskbarSymlink "$appsDir"/YACReaderLibrary
 }
```

Reply:
let's keep the current layout. such changes should be handled in a separate issue, if ever.